src/main/kotlin/com/avail/descriptor/representation/AbstractDescriptor.kt

```diff
@@ -0,0 +1,4202 @@
+	 */
+	@ThreadSafe
+	open fun printObjectOnAvoidingIndent (
+		self: AvailObject,
+		builder: StringBuilder,
+		recursionMap: IdentityHashMap<A_BasicObject, Void>,
+		indent: Int) =
+	with(builder) {
+		append('a')
+		val className = javaClass.simpleName
+		val shortenedName = className.substring(
+			0,
+			className.length - 10)
+		when (shortenedName.codePointAt(0))
+		{
+			'A'.toInt(),
+			'E'.toInt(),
+			'I'.toInt(),
+			'O'.toInt(),
+			'U'.toInt() ->
+				append('n')
+			else -> {}
+		}
+		append(' ')
+		append(shortenedName)
+		if (isMutable)
+		{
+			// Circled Latin capital letter M.
+			append('\u24C2')
+		}
+		else if (isShared)
+		{
+			// Circled Latin capital letter S.
+			append('\u24C8')
+		}
+		val cls: Class<Descriptor> = cast(this.javaClass)
+		val loader = cls.classLoader
+		val intEnumClass: Class<out IntegerSlotsEnum>? = nullableCast(
+			try
+			{
+				loader.loadClass("${cls.canonicalName}\$IntegerSlots")
+			}
+			catch (e: ClassNotFoundException)
+			{
+				null
+			})
+		val intSlots: Array<out IntegerSlotsEnum> =
+			if (intEnumClass != null) intEnumClass.enumConstants
+			else arrayOf()
+		run {
+			var i = 1
+			val limit = self.integerSlotsCount()
+			while (i <= limit)
+			{
+				val ordinal = min(i, intSlots.size) - 1
+				val slot = intSlots[ordinal]
+				if (getAnnotation(
+						slot as Enum<*>,
+						HideFieldInDebugger::class.java) == null
+					&& getAnnotation(
+						slot as Enum<*>,
+						HideFieldJustForPrinting::class.java) == null)
+				{
+					newlineTab(builder, indent)
+					val slotName = Nulls.stripNull(slot.fieldName())
+					val bitFields = bitFieldsFor(slot)
+					if (slotName[slotName.length - 1] == '_')
+					{
+						val subscript = i - intSlots.size + 1
+						append(slotName, 0, slotName.length - 1)
+						append('[')
+						append(subscript)
+						append(']')
+					}
+					else
+					{
+						val value = self.slot(slot)
+						if (bitFields.isEmpty())
+						{
+							append(slotName)
+							append(" = ")
+							append(value)
+						}
+						else
+						{
+							describeIntegerSlot(
+								self, value, slot, bitFields, builder)
+						}
+					}
+				}
+				i++
+			}
+		}
+		val objectEnumClass: Class<out ObjectSlotsEnum>? = nullableCast(
+			try
+			{
+				loader.loadClass("${cls.canonicalName}\$ObjectSlots")
+			}
+			catch (e: ClassNotFoundException)
+			{
+				null
+			})
+		val objectSlots: Array<out ObjectSlotsEnum> =
+			if (objectEnumClass != null) objectEnumClass.enumConstants
+			else arrayOf()
+		var i = 1
+		val limit = self.objectSlotsCount()
+		while (i <= limit)
+		{
+			val ordinal = min(i, objectSlots.size) - 1
+			val slot = objectSlots[ordinal]
+			if (getAnnotation(
+					slot as Enum<*>,
+					HideFieldInDebugger::class.java) == null
+				&& getAnnotation(
+					slot as Enum<*>,
+					HideFieldJustForPrinting::class.java) == null)
+			{
+				newlineTab(builder, indent)
+				val slotName = Nulls.stripNull(slot.fieldName())
```

Comment:
!!

Reply:
Turns out I don't even need `!!`.

Reply:
I will look for `Nulls` usages in Kotlin and remove them systematically.